Features/bootstrap/MainContext.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+use QATools\BehatExtension\Context\QAToolsContext;
+
+class MainContext extends QAToolsContext
+{
+
+	/**
+	 * @Given I'm on ':page'
+	 */
+	public function openPage($page)
+	{
+		$this->page = $this->pageFactory->getPage($page);
+		$this->page->open();
+	}
+
+	/**
+	 * @When I login with ':username' and ':password'
+	 */
+	public function login($username, $password)
+	{
+		$this->page->login($username, $password);
+	}
+
+	/**
+	 * @Then I should be on ':page'
+	 */
+	public function pageOpen($page)
+	{
+		$this->page = $this->pageFactory->getPage($page);
```

Comment:
> I haven't seen any best practices yet where to assign what.

If you'll find them anywhere (e.g. in different programming language/project), then let me know. I also want to know how this should be solved.

>  I just did it here cause I didn't know any good way to detect the successful or failing login in login that why I used @Then to fetch and store the correct page.

My idea was to assign `$this->page` in `@When` because that matches to user action (when user opens the page matches logically to page property assignment).